### ISSUE TITLE
Comments can become accidental doc comments

### ISSUE THREAD

**Issue by chrysn** [immunant/c2rust#310]
Some C comments are turned into doc comments inadvertently, resulting in "error[E0585]: found a documentation comment that doesn't document anything" or "error[E0658]: attributes on expressions are experimental" errors.

Minmal example:

```c
void empty(void) {
  /** This is deliberately empty *//** **/
}
```

results in

```rust
pub unsafe extern "C" fn empty() {
    /* * This is deliberately empty *//** **/
}
```

where (while not caught by this particular syntax highlighter) the second comment is a doc comment.

C comment styles are diverse, and (while not in this minimal form) things like this happen in eg. [in SiliconLabs' emlib (mirrored on github)](https://github.com/basilfx/RIOT-gecko-sdk/blob/31a7f247e19fca16dc961427c53f27f777f7b557/dist/emlib/inc/em_usart.h#L956).

On the road to fixing this, a first step is adding a test case (which can fortunately be included in the existing comments test trivially), then to hunt for where the regular replacemtent of `/**` with `/* *` happens and why it does not trigger on the second occurrence (maybe just a global replacement thing?).
